extensions/mssql/test/unit/tableExplorerWebViewController.test.ts

```diff
@@ -410,12 +410,19 @@ suite("TableExplorerWebViewController - Reducers", () => {
             // Assert
             // New row should be removed from newRows tracking array
             expect(controller.state.newRows).to.have.length(0);
-            // Row should still be in resultSet (just marked for deletion)
-            expect(controller.state.resultSet?.rowCount).to.equal(3);
-            expect(controller.state.resultSet?.subset).to.have.length(3);
-            // Row should be tracked in deletedRows array
-            expect(controller.state.deletedRows).to.include(100);
-            expect(controller.state.deletedRows).to.have.length(1);
+            // Row should be completely removed from resultSet (not just marked for deletion)
+            // because the backend completely removes newly created rows
+            expect(controller.state.resultSet?.rowCount).to.equal(2);
+            expect(controller.state.resultSet?.subset).to.have.length(2);
+            // Row should NOT be tracked in deletedRows (it's gone, not pending deletion)
+            expect(controller.state.deletedRows).to.not.include(100);
+            expect(controller.state.deletedRows).to.have.length(0);
+            // Should show "Row deleted" message instead of "Row marked for removal"
+            expect(
+                showInformationMessageStub.calledOnceWith(
+                    LocConstants.TableExplorer.rowDeletedSuccessfully,
+                ),
+            ).to.be.true;
```

Comment:
these comments would be even better as strings in the `expect()` call so that the output contains them if the test fails

Reply:
I'll address these in a follow up PR. Just want to get grid changes in place so I can start bringing in copy and export functionality in.